### PR TITLE
Save app state on pause

### PR DESCRIPTION
**Pull request by dorinagy**

### CODE REVIEW

File: app/src/main/java/com/greenfox/opal/gitinder/MainActivity.java [hunk 1]
```diff
@@ -49,6 +52,27 @@ protected void onCreate(Bundle savedInstanceState) {
     checkLogin();
   }
 
+  @Override
+  protected void onPause() {
+    saveOnPause(APP_STATE);
+    super.onPause();
```
Comment (adamgyulavari) [inline app/src/main/java/com/greenfox/opal/gitinder/MainActivity.java]: @dorinagy why the super is the second?

File: app/src/main/java/com/greenfox/opal/gitinder/MainActivity.java [hunk 2]
```diff
@@ -49,6 +52,27 @@ protected void onCreate(Bundle savedInstanceState) {
     checkLogin();
   }
 
+  @Override
+  protected void onPause() {
+    saveOnPause(APP_STATE);
```
Comment (adamgyulavari) [inline app/src/main/java/com/greenfox/opal/gitinder/MainActivity.java]: @dorinagy why passing the `APP_STATE`?

File: app/src/main/java/com/greenfox/opal/gitinder/MainActivity.java [hunk 3]
```diff
@@ -49,6 +52,27 @@ protected void onCreate(Bundle savedInstanceState) {
...
+  }
+
+  @Override
+  protected void onStop() {
+    saveOnPause(APP_STATE);
+    super.onStop();
+  }
+
+  public void saveOnPause(String appState) {
+    GitinderApp.app().basicComponent().inject(this);
```
Comment (adamgyulavari) [inline app/src/main/java/com/greenfox/opal/gitinder/MainActivity.java]: @dorinagy why injecting here?

File: app/src/test/java/com/greenfox/opal/gitinder/RobolectricTest.java [hunk 4]
```diff
@@ -44,4 +46,10 @@ public void checkLoginNoUser() throws Exception {
     Intent expectedIntent = new Intent(loginActivity, LoginActivity.class);
     assertEquals(expectedIntent.getClass(), shadowOf(mainActivity).getNextStartedActivity().getClass());
   }
+
+  @Test
+  public void saveStatusOnPause() {
+    mainActivity.onPause();
+    assertEquals(mainActivity.timestamp, mainActivity.preferences.getString("AppState", ""));
```
Comment (adamgyulavari) [inline app/src/test/java/com/greenfox/opal/gitinder/RobolectricTest.java]: @dorinagy use the constant key here as well

**Review by adamgyulavari — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/greenfox/opal/gitinder/MainActivity.java [hunk 5]
```diff
@@ -54,21 +54,20 @@ protected void onCreate(Bundle savedInstanceState) {
 
   @Override
   protected void onPause() {
-    saveOnPause(APP_STATE);
+    saveOnPause();
```
Comment (adamgyulavari) [inline app/src/main/java/com/greenfox/opal/gitinder/MainActivity.java]: @dorinagy its still the second

**Review by adamgyulavari — CHANGES_REQUESTED** [review]

File: app/src/test/java/com/greenfox/opal/gitinder/RobolectricTest.java [hunk 6]
```diff
@@ -44,4 +46,10 @@ public void checkLoginNoUser() throws Exception {
     Intent expectedIntent = new Intent(loginActivity, LoginActivity.class);
     assertEquals(expectedIntent.getClass(), shadowOf(mainActivity).getNextStartedActivity().getClass());
   }
+
+  @Test
+  public void saveStatusOnPause() {
+    mainActivity.onPause();
```
Comment (adamgyulavari) [inline app/src/test/java/com/greenfox/opal/gitinder/RobolectricTest.java:51]: @dorinagy use roboeletric shadow for pausing

**Review by adamgyulavari — CHANGES_REQUESTED** [review]

**Comment by adamgyulavari** [comment]
@dorinagy and resolve conflicts

**Review by balintvecsey — APPROVED** [review]

**Pull request merged**